Source/Diagnostics/Diagnostics.cpp

```diff
@@ -251,7 +251,124 @@ Diagnostics::InitBaseData ()
         m_flush_format = std::make_unique<FlushFormatCheckpoint>() ;
     } else if (m_format == "ascent"){
         m_flush_format = std::make_unique<FlushFormatAscent>();
-    } else if (m_format == "sensei"){
+        std::cout<< " m_diag_name " <<m_diag_name<<"\n";
+        amrex::ParmParse pp_diag_viz(m_diag_name);
+        int write_yaml_file = 0;
+        pp_diag_viz.query("viz", write_yaml_file);
+        if (write_yaml_file) {// && amrex::ParallelDescriptor::IOProcessor()){
```

Comment:
Only process 0 writes:
```suggestion
        if (write_yaml_file && amrex::ParallelDescriptor::IOProcessor()){
```